logbus/setup/setup.go

```diff
@@ -122,40 +122,39 @@ func (bs *BusSetup) DumpManifestToFile(path string) error {
 	return nil
 }
 
-// Close closes the BusSetup.
+// Close the bus setup & gather all errors.
 func (bs *BusSetup) Close() error {
-	var retErr error
-	errs := bs.ConsoleWriter.Errors()
-	var cwErr error
-	for _, err := range errs {
-		cwErr = multierror.Append(cwErr, err)
-	}
-	if cwErr != nil {
-		retErr = multierror.Append(retErr, errors.Wrap(cwErr, "console writer"))
+	var ret error
+
+	if errs := bs.ConsoleWriter.Errors(); len(errs) > 0 {
+		multi := &multierror.Error{Errors: errs}
+		ret = multierror.Append(ret, errors.Wrap(multi, "console writer"))
 	}
-	fErr := bs.Formatter.Close()
-	if fErr != nil {
-		retErr = multierror.Append(retErr, errors.Wrap(fErr, "formatter"))
+
+	if err := bs.Formatter.Close(); err != nil {
+		ret = multierror.Append(ret, errors.Wrap(err, "formatter"))
 	}
+
 	if bs.BusDebugWriter != nil {
-		errs := bs.BusDebugWriter.Errors()
-		var bdwErr error
-		for _, err := range errs {
-			bdwErr = multierror.Append(bdwErr, err)
-		}
-		if bdwErr != nil {
-			retErr = multierror.Append(retErr, errors.Wrap(bdwErr, "bus debug writer"))
+		if errs := bs.BusDebugWriter.Errors(); len(errs) > 0 {
+			multi := &multierror.Error{Errors: errs}
+			ret = multierror.Append(ret, errors.Wrap(multi, "bus debug writer"))
 		}
 	}
+
 	if bs.LogStreamer != nil {
-		manifestsWritten, logsWritten, err := bs.LogStreamer.Close()
-		if err != nil {
-			retErr = multierror.Append(retErr, errors.Wrap(err, "log streamer"))
+		bs.LogStreamer.Close()
+		if errs := bs.LogStreamer.Errs(); len(errs) > 0 {
+			multi := &multierror.Error{}
+			for _, err := range errs {
+				streamErr := &cloud.StreamError{}
+				if bs.verbose || (errors.As(err, &streamErr) && !streamErr.Recoverable) {
```

Comment:
What would happen if `errors.As(err, &streamErr)` fails? This feels like a potential for a different error type to end up here, which would then be ignored.

Currently the code shouldn't allow this to happen, but this feels fragile as a future code change could return a different error type without knowing this restriction.

We should either append it, or panic().